analysis/import/SonarImporter/src/main/java/de/maibornwolff/codecharta/importer/sonar/dataaccess/SonarMetricsAPIDatasource.java

```diff
@@ -59,17 +63,36 @@ public SonarMetricsAPIDatasource(String user, URL baseUrl) {
         this.baseUrl = baseUrl;
     }
 
-    public List<MetricObject> getAvailableMetrics() {
+    public List<String> getAvailableMetricsList() {
+        int noPages = getNumberOfPages();
+
+        return Flowable.range(1, noPages)
+                .flatMap(p -> Flowable.just(p)
+                        .subscribeOn(Schedulers.io())
+                        .map(this::getAvailableMetrics))
+                .filter(m -> m.getMetrics() != null)
+                .flatMap(m -> Flowable.fromIterable(m.getMetrics()))
+                .filter(m -> !m.isHidden() && m.isFloatType())
+                .map(MetricObject::getKey).distinct().toSortedList().blockingGet();
+    }
+
+    public Metrics getAvailableMetrics(int page) {
 
         Client client = ClientBuilder.newClient();
         client.register(ErrorResponseFilter.class);
         client.register(GsonProvider.class);
 
-        Invocation.Builder request = client.target(baseUrl + METRICS).request();
+        String url = String.format(METRICS_URL_PATTERN, baseUrl, page);
+        Invocation.Builder request = client.target(url).request();
         if (!user.isEmpty()) {
             request.header("Authorization", "Basic " + AuthentificationHandler.createAuthTxtBase64Encoded(user));
         }
-        Metrics metric = request.get(Metrics.class);
-        return metric.getMetrics();
+        return request.get(Metrics.class);
+    }
+
+
+    public int getNumberOfPages() {
```

Comment:
this is strange. getAvailableMetrics(int page) takes the index of a page. So why would the method which returns the number of pages call getAvailableMetrics(1). That seems like a bug.